department-of-veterans-affairs/va.gov-cms
# VACM-9512 sitewide alert types

## Description

Closes #9512, #8299, #8300. Styles the sitewide alerts based on the figma designs seen in 9512/8299. This module has dismissible behavior included. Adds these different alerts as a component to storybook, uses that in the drupal theme.

## Testing done
- Verified that site alerts rendered with proper styles in the 3 alert types.
- Verified that dismissed alerts stay dismissed and can be un-dismissed with appropriate fields checked on form updates.
- Verified that the site shield alert blocks the full page.

## Screenshots
storybook:
<img width="118" alt="Screen Shot 2022-07-06 at 4 42 34 PM" src="https://user-images.githubusercontent.com/11279744/177660762-3311ce89-c624-4025-8523-12fd6b2d551d.png">

informational, dismissible:
<img width="758" alt="Screen Shot 2022-07-06 at 4 41 05 PM" src="https://user-images.githubusercontent.com/11279744/177660784-10d77380-a286-44ce-8f6f-0546f2008222.png">

warning, dismissible:
<img width="1512" alt="Screen Shot 2022-07-06 at 4 41 34 PM" src="https://user-images.githubusercontent.com/11279744/177660785-037463c1-4dd7-46ff-9898-6d28504c9482.png">

site shield, non-dismissible:
<img width="757" alt="Screen Shot 2022-07-06 at 4 41 52 PM" src="https://user-images.githubusercontent.com/11279744/177660786-bb87fd30-af9a-4471-b095-3edfdcf66560.png">



## QA steps
- In the storybook link generated below, see that the "Sitewide Alert" component exists in the list with variations.

- Then on the environment generated below, log in as an admin user and navigate to `/admin/content/sitewide_alert`. Create a new alert.
- Start with "Informational" style, and check the box to make it dismissible. Alert Title is optional. Save your alert and see it appear on the page.
- Dismiss the alert, then edit it.
- Change the style to "Warning", and check the box to ignore previous dismissals. Save the alert.
- See that the alert has changed to yellow, and is dismissible again.
- Dismiss the alert, then edit it.
- Change the style to "Site Shield" and uncheck it as dismissible. Save the alert.
- See that the "Site Shield" style functions in the same manner as our existing site shields (full page, can scroll behind but can't click anything).
- optional: inspect and delete the site alert container so that you can edit it again and switch it off of site shield / make it inactive.

### Definition of Done


### Select Team for PR review


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
